### ISSUE TITLE
duplicate-heading false positive: #[tokio::main] inside fenced code block is detected as heading

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#330]
## Bug Report

`duplicate-heading` rule incorrectly flags `#[tokio::main]` inside a fenced code block as a duplicate heading.

## Steps to Reproduce

Create a Markdown file containing `#[tokio::main]` in two separate fenced code blocks:

~~~markdown
## Full source

```rust
#[tokio::main]
async fn main() {
    // ...
}
```

## Closer look

```rust
#[tokio::main]
async fn main() {
```
~~~

Run gomarklint:

```sh
gomarklint file.md
```

## Expected Behavior

No errors. `#[tokio::main]` is inside a fenced code block and should not be treated as a Markdown heading.

## Actual Behavior

```
file.md:10: [error] duplicate heading: "[tokio::main]"
```

## Environment

- gomarklint version: v3.2.3
- OS: macOS 14 (arm64)

## Workaround

Using disable comments to suppress the false positive:

```markdown
<!-- gomarklint-disable duplicate-heading -->

```rust
#[tokio::main]
async fn main() {
```

```

This works but adds noise to published content.

## Root Cause (hypothesis)

The heading scanner appears to scan lines starting with `#` without first checking whether the current position is inside a fenced code block.
